os/using-systemd-and-udev-rules.md

```diff
@@ -0,0 +1,74 @@
+# Using systemd and udev Rules 
+
+In our example we will use libvirt VM. First of all we have to create systemd unit file `/etc/systemd/system/test.service`:
+
+```
+[Service]
+Type=oneshot
+ExecStart=/usr/bin/echo 'device has been attached'
+```
+
+Then we have to start `udevadm monitor --environment` to monitor kernel events.
+
+Once you've attached virtio libvirt device (i.e. `virsh attach-disk coreos /dev/VG/test vdc`) you'll see similar udev output:
+
+```
+UDEV  [545.954641] add      /devices/pci0000:00/0000:00:18.0/virtio4/block/vdb (block)
+.ID_FS_TYPE_NEW=
+ACTION=add
+DEVNAME=/dev/vdb
+DEVPATH=/devices/pci0000:00/0000:00:18.0/virtio4/block/vdb
+DEVTYPE=disk
+ID_FS_TYPE=
+MAJOR=254
+MINOR=16
+SEQNUM=1327
+SUBSYSTEM=block
+SYSTEMD_WANTS=test.service
+TAGS=:systemd:
+USEC_INITIALIZED=545954447
+```
+
+Accoring to text beyond udev rule should look this way:
+
+```
+ACTION=="add", SUBSYSTEM=="block", TAG+="systemd", ENV{SYSTEMD_WANTS}="test.service"
+```
+
+Now when we use this command `virsh attach-disk coreos /dev/VG/test vdc` on host machine, we should see `device has been attached` message inside CoreOS node.
+
+## Cloud-Config example
+
+Cloud-Config example should look this way:
```

Comment:
"To use the unit and udev rule with cloud-config, modify this example as needed:"